solr/core/src/java/org/apache/solr/cloud/api/collections/DeleteCollectionCmd.java

```diff
@@ -159,6 +165,36 @@ public void call(ClusterState state, ZkNodeProps message, NamedList results) thr
         });
       }
 
+      // remove auto-created policy if this is the only collection that uses it
```

Comment:
Maybe we should disallow people from using these auto created policies in the first place?